src/library_dylink.js

```diff
@@ -270,52 +270,118 @@ var LibraryDylink = {
+      populateCustomSection(dylinkSectionInt8);
+      return customSection;
+    }
+    else{
+      var int32View = new Uint32Array(new Uint8Array(binary.subarray(0, 24)).buffer);
+      assert(int32View[0] == 0x6d736100, 'need to see wasm magic number'); // \0asm
+      // we should see the dylink section right after the magic number and wasm version
+      assert(binary[8] === 0, 'need the dylink section to be first')
+      next = 9;
+      getLEB(binary); //section size
+      assert(binary[next] === 6);                 next++; // size of "dylink" string
+      assert(binary[next] === 'd'.charCodeAt(0)); next++;
+      assert(binary[next] === 'y'.charCodeAt(0)); next++;
+      assert(binary[next] === 'l'.charCodeAt(0)); next++;
+      assert(binary[next] === 'i'.charCodeAt(0)); next++;
+      assert(binary[next] === 'n'.charCodeAt(0)); next++;
+      assert(binary[next] === 'k'.charCodeAt(0)); next++;
+
+      populateCustomSection(binary);
+      return customSection;
     }
+  },
+
+  // Module.symbols <- libModule.symbols (flags.global handler)
+  $mergeLibSymbols__deps: ['$asmjsMangle'],
+  $mergeLibSymbols: function(libModule) {
+    // add symbols into global namespace TODO: weak linking etc.
+    for (var sym in libModule) {
+      if (!libModule.hasOwnProperty(sym)) {
+        continue;
+      }
 
+      // When RTLD_GLOBAL is enable, the symbols defined by this shared object will be made
+      // available for symbol resolution of subsequently loaded shared objects.
+      //
+      // We should copy the symbols (which include methods and variables) from SIDE_MODULE to MAIN_MODULE.
+
+      var module_sym = asmjsMangle(sym);
+
+      if (!Module.hasOwnProperty(module_sym)) {
+        Module[module_sym] = libModule[sym];
+      }
+#if ASSERTIONS == 2
+      else {
+        var curr = Module[sym], next = libModule[sym];
+        // don't warn on functions - might be odr, linkonce_odr, etc.
+        if (!(typeof curr === 'function' && typeof next === 'function')) {
+          err("warning: symbol '" + sym + "' from '" + lib + "' already exists (duplicate symbol? or weak linking, which isn't supported yet?)"); // + [curr, ' vs ', next]);
+        }
+      }
+#endif
+    }
+  },
+
+  // Loads a side module from binary data or compiled Module. Returns the module's exports or a
+  // promise that resolves to its exports if the loadAsync flag is set.
+  $loadWebAssemblyModule__deps: ['$loadDynamicLibrary', '$createInvokeFunction', '$getMemory', '$relocateExports', '$resolveGlobalSymbol', '$GOTHandler', '$getSideModuleCustomSection'],
+  $loadWebAssemblyModule: function(binary, flags) {
+    var customSection = getSideModuleCustomSection(binary);
+    var memorySize = customSection.memorySize;
+    var memoryAlign = customSection.memoryAlign;
+    var tableSize = customSection.tableSize;
+    var tableAlign = customSection.tableAlign;
+    var neededDynlibs = customSection.neededDynlibs;
```

Comment:
Perhaps just use the `customSection` directly rather than create five additional locals?  Perhaps call it `metadata` or `dylinkInfo` or something with a little more meaning?

Reply:
unfortunately there are nested functions which use the locals in the parent function. it would require quite abit of rewrite without these.

Reply:
Can you make a followup to clean this up?

Also can you call this variable `metadata` rather than `customSection`